### PR TITLE
provider/appengine upsert load balancer operation

### PR DESCRIPTION
**Pull request by danielpeach**
@lwander or @duftler please review

It's not really an upsert operation, just an update.

### CODE REVIEW

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    def credentials = description.credentials
+    def appengine = credentials.appengine
+    def loadBalancerName = description.loadBalancerName
+    def updateSplit = description.split
+
+    // TODO(dpeach): do validation for traffic migration.
+    def migrateTraffic = description.migrateTraffic
+
+    if (updateSplit.allocations) {
+      validateServerGroupsCanBeEnabled(updateSplit.allocations, loadBalancerName)
```
Comment (lwander) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy]: Why not have this in your validator?

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    def override = copyAndOverrideAncestorSplit(ancestorLoadBalancer.split, updateSplit)
+
+    def service = new Service(split: new TrafficSplit(allocations: override.allocations, shardBy: override.shardBy.toString()))
+
+    appengine.apps().services().patch(credentials.project, loadBalancerName, service)
+      .setUpdateMask("split")
+      .setMigrateTraffic(migrateTraffic)
+      .execute()
+
+    // TODO(dpeach): write operation poller and poll for operation completion here.
```
Comment (lwander) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy]: I don't think you need an operation poller here, it's up to Orca to determine when this has finished upserting.

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      validateServerGroupsCanBeEnabled(updateSplit.allocations, loadBalancerName)
+    }
+
+    def ancestorLoadBalancer = appEngineLoadBalancerProvider.getLoadBalancer(credentials.name, loadBalancerName)
+    def override = copyAndOverrideAncestorSplit(ancestorLoadBalancer.split, updateSplit)
+
+    def service = new Service(split: new TrafficSplit(allocations: override.allocations, shardBy: override.shardBy.toString()))
+
+    appengine.apps().services().patch(credentials.project, loadBalancerName, service)
+      .setUpdateMask("split")
```
Comment (lwander) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy]: Would it make sense to have this be user-configurable?

Comment (danielpeach) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy]: `split` is the only field on a service that can be updated (even though the split has many sub properties).

**Pull request merged**